### PR TITLE
Added note about "stacked usings" and highlight lines.

### LINKED ISSUE #23202: Define "stacked usings"
What exactly does *stacking* mean? I see 2 possibilities:

1. Constructing multiple objects and calling await afterwards
2. Mixing the implicit and explicit forms of the using statement

Alternatively, how acceptable would the following example be? It differs from the *nested* example by constructing objTwo before the awaits:

	using System;
	using System.Threading.Tasks;

	class ExampleOneProgram
	{
		static async Task Main()
		{
			var objOne = new ExampleAsyncDisposable();
			// Exception thrown on .ctor
			var objTwo = new ExampleAsyncDisposable();
			
			await using (objOne.ConfigureAwait(false))
			{
				// Interact with the objOne instance.

				await using (objTwo.ConfigureAwait(false))
				{
					// Interact with the objOne and/or objTwo instance(s).
				}
			}

			Console.ReadLine();
		}
	}

I believe that **2** is the correct answer, but I still don't see how and why would the implicit using's Dispose call be skipped when the *nesting* case manages to call the explicit one's properly.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 61a961dc-3ebf-2a96-a6eb-2e0b62ab909f
* Version Independent ID: d72276e1-9059-815f-dc14-a03a7544ab87
* Content: [Implement a DisposeAsync method](https://docs.microsoft.com/en-us/dotnet/standard/garbage-collection/implementing-disposeasync)
* Content Source: [docs/standard/garbage-collection/implementing-disposeasync.md](https://github.com/dotnet/docs/blob/main/docs/standard/garbage-collection/implementing-disposeasync.md)
* Product: **dotnet-fundamentals**
* GitHub Login: @IEvangelist
* Microsoft Alias: **dapine**

### PR DESCRIPTION
**Pull request by IEvangelist**
## Summary

Added note about "stacked usings" and highlight lines.

Fixes #23202

### CODE REVIEW

File: docs/standard/garbage-collection/implementing-disposeasync.md [hunk 1]
```diff
@@ -120,7 +120,9 @@ In the preceding example, each asynchronous clean up operation is implicitly sco
 
 If an exception is thrown from the `AnotherAsyncDisposable` constructor, then `objOne` does not get properly disposed:
```
Comment (gewarren) [inline docs/standard/garbage-collection/implementing-disposeasync.md:121]:
```suggestion
The highlighted lines in the following code show what it means to have "stacked usings". If an exception is thrown from the `AnotherAsyncDisposable` constructor, then `objOne` does not get properly disposed.
```

File: docs/standard/garbage-collection/implementing-disposeasync.md [hunk 2]
```diff
@@ -120,7 +120,9 @@ In the preceding example, each asynchronous clean up operation is implicitly sco
...
 
-:::code language="csharp" id="dontdothis" source="../../../samples/snippets/csharp/VS_Snippets_CLR/conceptual.asyncdisposable/stacked-await-usings.cs":::
+:::code language="csharp" id="dontdothis" source="../../../samples/snippets/csharp/VS_Snippets_CLR/conceptual.asyncdisposable/stacked-await-usings.cs" highlight="9-10":::
+
+In the preceding code, the highlighted lines show what it means to have "stacked usings".
```
Comment (gewarren) [inline docs/standard/garbage-collection/implementing-disposeasync.md:125]:
```suggestion
```

**Review by gewarren — COMMENTED** [review]
Suggestion for line 99:

```
...it's possible that [stacking `await using` statements](#stacked-usings) with <xref:System.Threading.Tasks.ValueTask.ConfigureAwait%2A> could prevent calls...
```

**Review by BillWagner — APPROVED** [review]
This LGTM @IEvangelist 

You can :shipit: when ready.

**Pull request merged**